GLES2 scene shader compile failure when using shadows

**Godot version:**
45842c0ea9c6e80f56954efb5e08a7c81e372fc4

**OS/device including version:**
(Happens both in Desktop and WebGL)
Windows 10 geforce 635m crap drivers and Firefox/Chrome Angle dx11 HD4000

**Issue description:**
When enabling shadow casting (in this case with a spotlight) this error appears:

 drivers\gles2\shader_gles2.cpp:196 - SceneShaderGLES2: Fragment shader compilation failed:
0(1297) : error C1102: incompatible type for parameter #2 ("spos.88")
0(1297) : error C1104: too many parameters in function call

**Notes:**
actual line in scene.glsl is 1837:
		float shadow = sample_shadow(light_shadow_atlas, splane.xy, splane.z);

tried changing it to sample_shadow(light_shadow_atlas, splane), looks like it compiles, but dunno if it is correct

--- a/drivers/gles2/shaders/scene.glsl
+++ b/drivers/gles2/shaders/scene.glsl
@@ -1307,8 +1307,7 @@ LIGHT_SHADER_CODE
 #define SAMPLE_SHADOW_TEXEL(p_shadow, p_pos, p_depth) step(p_depth, texture2D(p_shadow, p_pos).r)
 #define SAMPLE_SHADOW_TEXEL_PROJ(p_shadow, p_pos) step(p_pos.z, texture2DProj(p_shadow, p_pos).r)
 
-float sample_shadow(
-		highp sampler2D shadow, highp vec4 spos) {
+float sample_shadow(highp sampler2D shadow, highp vec4 spos) {
 
 #ifdef SHADOW_MODE_PCF_13
 
@@ -1923,12 +1922,12 @@ FRAGMENT_SHADER_CODE
 		highp vec4 splane = shadow_coord;
 		splane.xyz /= splane.w;
 
-		float shadow = sample_shadow(light_shadow_atlas, splane.xy, splane.z);
+		float shadow = sample_shadow(light_shadow_atlas, splane);
 		light_att *= shadow;
 	}
 #endif
 
-#endif
+#endif // LIGHT_MODE_SPOT
 
 #ifdef USE_VERTEX_LIGHTING
 	//vertex lighting

